app/uk/gov/hmrc/testuser/services/Generator.scala

```diff
@@ -213,3 +214,38 @@ class PensionSchemeAdministratorIdentifierGenerator(random: Random = new Random)
     PensionSchemeAdministratorIdentifier(s"$initialCharacter$remainingDigits")
   }
 }
+
+object VrnChecksum {
+
+  def apply(s: String) = s + calcCheckSum97(weightedTotal(s))
+
+  def isValid(vrn: String): Boolean = {
+    if (regexCheck(vrn)) {
+      val total = weightedTotal(vrn)
+      val checkSumPart = takeCheckSumPart(vrn)
+      if (checkSumPart == calcCheckSum97(total).toInt) true
+      else checkSumPart == calcCheckSum9755(total)
+    } else false
+  }
+
+  private def calcCheckSum97(total: Int): String = {
+    var x = total
+    while (x >= 0) {
+      x = x - 97
+    }
+    if (x > -10) s"0${-x}" else (-x).toString
```

Comment:
What about replacing this line with something like:
f"${Math.abs(x)}%02d"

I think this is easier to see that you are formatting into a string of 2 digits, rather than trying to understand why there is a check for x being greater than -10.